src/XProfile/VirtoCommerce.XProfile/Queries/GetOrganizationByIdQueryHandler.cs

```diff
@@ -13,9 +13,9 @@ public GetOrganizationByIdQueryHandler(IOrganizationAggregateRepository organiza
             _organizationAggregateRepository = organizationAggregateRepository;
         }
 
-        public Task<OrganizationAggregate> Handle(GetOrganizationByIdQuery request, CancellationToken cancellationToken)
+        public async Task<OrganizationAggregate> Handle(GetOrganizationByIdQuery request, CancellationToken cancellationToken)
         {
-            return _organizationAggregateRepository.GetOrganizationByIdAsync(request.OrganizationId);
+            return (OrganizationAggregate)await _organizationAggregateRepository.GetMemberAggregateRootByIdAsync(request.OrganizationId);
```

Comment:
We can remove the async/await combination here if we remove the type conversion.